replica/database.cc

```diff
@@ -1427,6 +1427,10 @@ future<> database::create_in_memory_keyspace(const lw_shared_ptr<keyspace_metada
         // don't make system keyspace writes wait for user writes (if under pressure)
         kscfg.dirty_memory_manager = &_system_dirty_memory_manager;
     }
+    if (extensions().is_extension_internal_keyspace(ksm->name())) {
+        // don't make internal keyspaces write wait for user writes (if under pressure), and also to avoid possible deadlocks.
+        kscfg.dirty_memory_manager = &_system_dirty_memory_manager;
+    }
```

Comment:
I don't understand these extensions. Why are we extending our own stuff?

Reply:
@elcallio can you please answer?
What I could make out of it is that it is a way to deviate from the cassandra compatible schema.
This specific extentions purpose (internal_keyspace) is documented as:
```
/**
     * Adds a keyspace to "extension internal" set.
     *
     * Such a keyspace must be loaded before/destroyed after any "normal" user keyspace.
     * Thus a psuedo-system/internal keyspce.
     * This has little to no use in open source version, and is temporarily bridged with
     * the static version of same functionality in distributed loader. It is however (or will
     * be), essential to enterprise code. Do not remove.
     */
```

Reply:
Here is another refference to an issue that explains the motivation:
https://github.com/scylladb/scylladb/issues/13334

Reply:
Sorry, I missed the above comment request; @avikivity: We use "extensions" to bridge OSS/enterprise functionality, i.e. to avoid code skews across the versions. They are abstractions/black-box layers for special casing and opt-in features. 
(I feel this has been explained before, since you've obviously signed off on the whole approach once + all the various additions made here).

Reply:
We're using the extensions to talk to ourselves here.